design-docs/Codegen.md

```diff
@@ -6,9 +6,14 @@ For more complex queries, involving [merged fields](https://spec.graphql.org/dra
 
 * **responseBased**: the Kotlin models map the received json.
 * **operationBased**: the Kotlin models map the sent operation.
+* **experimental_operationBasedWithInterfaces**: based on **operationBased**, but expose more type information. 
 * **compat**: for compatibility with Apollo Kotlin 2.x
 
-`operationBased` will generate less code but will use more memory and expose less type information. `responseBased` will generate interfaces to access the models in a more polymorphic way. It will also store each merged field exactly once and have more efficient json parsing. That comes at the price of more generated code.
+`responseBased` will generate interfaces to access the models in a more polymorphic way. It will also store each merged field exactly once and have more efficient json parsing. That comes at the price of more generated code. 
+
+`operationBased` will generate less code but will use more memory and expose less type information.
+
+`experimental_operationBasedWithInterfaces` same as `operationBased` but will generate interfaces to access model similar to `responseBased`. 
```

Comment:
I went with 

```
generates interfaces for selection sets that contain fragments to make it easier to use `when` statements.
```

Because it's very far from `responseBased` codegen actualy.